components/projects/EnvironmentVariableEditor.tsx

```diff
@@ -0,0 +1,192 @@
+import { ComponentProps, useCallback, useRef, useState } from "react";
+import { parseEnvFile, parseStringEnvContents } from "@/utils/helpers";
+import { EyeIcon, XIcon } from "lucide-react";
+import { useDropzone } from "react-dropzone";
+import { Button, Container, InputGroup } from "@/components/theme";
+import clsx from "clsx";
+
+export interface KeyPair {
+  envKey: string;
+  envValue: string;
+}
+
+export function EnvironmentVariableEditor() {
+  const [envKeys, setEnvKeys] = useState<KeyPair[]>([]);
+  const pastingInputIndex = useRef(0);
+
+  const onDrop = useCallback((acceptedFiles: File[]) => {
+    const file = acceptedFiles[0];
+
+    parseEnvFile(file, (pairs) => {
+      setEnvKeys([...pairs]);
+    });
+  }, []);
+
+  const { getRootProps, getInputProps, isDragActive, open } = useDropzone({
+    noClick: true,
+    onDrop,
+  });
+
+  const handleAddMoreEnvClick = () => {
+    setEnvKeys([...(envKeys as KeyPair[]), { envKey: "", envValue: "" }]);
+  };
+  const handleRemoveEnvPairClick = (index: number) => {
+    setEnvKeys(envKeys?.filter((_, i) => i !== index));
+  };
+
+  const handlePaste = (event: any) => {
+    const pastedEnvKeyValuePairs = parseStringEnvContents(
+      event.clipboardData?.getData("text"),
+    );
+
+    if (
+      !(pastedEnvKeyValuePairs[0].envKey && pastedEnvKeyValuePairs[0].envValue)
+    ) {
+      return;
+    }
+
+    event.preventDefault();
+    const envKeysBeforePastingInput = envKeys.slice(
+      0,
+      pastingInputIndex.current,
+    );
+
+    const envKeysAfterPastingInput = envKeys.slice(
+      pastingInputIndex.current + 1,
+    );
+    setEnvKeys([
+      ...envKeysBeforePastingInput,
+      ...pastedEnvKeyValuePairs,
+      ...envKeysAfterPastingInput,
+    ]);
+  };
+
+  return (
+    <Container
+      className={`${
+        isDragActive ? "border-teal-300" : "border-darker"
```

Comment:
❤️ it, `clsx` can come handy here as well

Reply:
done 😀